lib/syscall_shim/include/uk/syscall.h

```diff
@@ -61,32 +61,87 @@ extern "C" {
 #endif /* CONFIG_LIBSYSCALL_SHIM && CONFIG_LIBSYSCALL_SHIM_NOWRAPPER */
 #endif /* UK_LIBC_SYSCALLS */
 
+#ifdef CONFIG_LIBSYSCALL_SHIM
+/**
+ * Library-internal thread-local variable containing the return address
+ * of the caller of the currently called system call.
+ * NOTE: Use the `uk_syscall_return_addr()` macro to retrieve the return address
+ *       within a system call implementation. Please note that this macro is
+ *       only available if `lib/syscall_shim` is part of the build.
+ *       This implies that every system call implementation that require to know
+ *       the return address have a dependency to `lib/syscall_shim`.
+ */
+extern __uk_tls __uptr _uk_syscall_return_addr;
+
+/**
+ * Returns the return address of the currently called system call.
+ */
+#define uk_syscall_return_addr()					\
+	((__uptr)((_uk_syscall_return_addr != 0x0)			\
+		  ? _uk_syscall_return_addr : __return_addr(0)))	\
+
+#define __UK_SYSCALL_RETADDR_ENTRY();					\
+	do { _uk_syscall_return_addr = uk_syscall_return_addr(); } while (0)
+#define __UK_SYSCALL_RETADDR_CLEAR();					\
+	do { _uk_syscall_return_addr = 0x0; } while (0)
+
+#else /* !CONFIG_LIBSYSCALL_SHIM */
+
+/* NOTE: The usage of `uk_syscall_return_addr()` requires the `syscall_shim`
+ *       library to be present. In order to avoid mis-configurations, we
+ *       inject an always failing compile-time assertion to stop compilation.
+ */
```

Comment:
```suggestion
/**
 * NOTE: The usage of `uk_syscall_return_addr()` requires the `syscall_shim`
 *       library to be present. In order to avoid mis-configurations, we
 *       inject an always failing compile-time assertion to stop compilation.
 */
```

When should we use /** vs /*?

Reply:
Normally `/**` should be used by Doxygen and to describe interfaces. Since this is a normal comment block, I'd keep the single-asterix version: `/*`. If you think it helps readability, I am applying your suggestion but use `/*`.